### PR TITLE
Add group and inventory / Extract common methods into a mixin

### PR DESCRIPTION
**Pull request by syncrou**

### CODE REVIEW

**Comment by syncrou** [comment]
@bdunne - Please review

File: lib/ansible_tower_client/host.rb [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 
-    def initialize(raw_host)
-      @id            = raw_host["id"]
-      @name          = raw_host["name"]
-      @instance_id   = raw_host["instance_id"]
-      @url           = raw_host["url"]
-      @raw_host_body = raw_host
+    attr_reader :id, :name, :url, :raw_body
+
+    def initialize(raw_body)
```
Comment (bdunne) [inline lib/ansible_tower_client/host.rb]: You could probably set `@instance_id` then super.

File: lib/ansible_tower_client/instance_methods.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module AnsibleTowerClient
+  module InstanceMethods
+    attr_reader :id, :name, :url, :raw_body
+
+    def initialize(raw_body)
+      @id   = raw_body["id"]
```
Comment (bdunne) [inline lib/ansible_tower_client/instance_methods.rb]: Can you align on `=`?

File: lib/ansible_tower_client/host.rb [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
-    attr_reader :raw_host_body
+    include InstanceMethods
 
-    def initialize(raw_host)
-      @id            = raw_host["id"]
-      @name          = raw_host["name"]
-      @instance_id   = raw_host["instance_id"]
-      @url           = raw_host["url"]
-      @raw_host_body = raw_host
+    attr_reader :id, :name, :url, :raw_body
```
Comment (bdunne) [inline lib/ansible_tower_client/host.rb]: Don't forget `:instance_id`

File: lib/ansible_tower_client/collection_methods.rb [hunk 4]
```diff
@@ -13,7 +13,7 @@ def all
     end
 
     def find(id)
-      body = JSON.parse(Api.get("#{self.endpoint}/#{id}/").body)
+      body = JSON.parse(Api.get("#{endpoint}/#{id}/").body)
       raise NoMethodError if body['id'].nil?
```
Comment (bdunne) [inline lib/ansible_tower_client/collection_methods.rb]: Sorry, I should have mentioned this on the other PR... I think `NoMethodError` is strange to use here.  Maybe we need a `ResourceNotFoundError` or something like that.  Does Faraday have anything like this?

Comment (syncrou) [inline lib/ansible_tower_client/collection_methods.rb]: I'll take a look at Faraday's lists of exceptions.

**Comment by syncrou** [comment]
@bdunne - Please review.

**Pull request merged**